### ISSUE TITLE
on adding entries and data sets in the menu's add Dynamical data, an error occurred and the app stops

### ISSUE THREAD

**Issue by codaphrenic** [PhilJay/MPAndroidChart#4023]
**Summary**

When I use the "Add Entry" feature then I hesitate and I want to delete the entry via the "Remove Entry" feature. But the display entry is not lost on the screen. To shorten, I use the "Clear Chart" feature. But the app stops automatically. I do it repeatedly, unfortunately the same problem happens again.

**Expected Behavior**

Want to add an entry and dataset in the Dynamical data adding menu.

**Steps to Reproduce**

Open the menu "Dynamical Data Adding". then click the "Add Entry" feature to add an entry. The next step is to delete the entry using the "Remove Entry" feature then you can not change anything yet. Please use the "Clear Chart" feature to remove it. When you finish deleting all, click the "Add Entry" feature again to add the entry. Then the app will stop automatically.

**Device**
 - Device: Redmi Note 5A
 - Android Version: 7.1.9
 - Library Version: 3.0.3

Below is the crash log -

```
FATAL EXCEPTION: main
Process: com.xxmassdeveloper.mpchartexample, PID: 32230
java.lang.NullPointerException: Attempt to invoke virtual method 
'com.github.mikephil.charting.interfaces.datasets.IDataSet com.github.mikephil.charting.data.ChartData.getDataSetByIndex(int)' on a null object reference
    at com.xxmassdeveloper.mpchartexample.DynamicalAddingActivity.addEntry
(DynamicalAddingActivity.java:54)
    at com.xxmassdeveloper.mpchartexample.DynamicalAddingActivity.onOptionsItemSelected
(DynamicalAddingActivity.java:169)
    at android.app.Activity.onMenuItemSelected(Activity.java:3210)
    at android.support.v4.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:406)
    at com.android.internal.policy.PhoneWindow.onMenuItemSelected(PhoneWindow.java:1237)
    at com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected
(MenuBuilder.java:766)
    at com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:152)
    at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:909)
    at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:899)
    at com.android.internal.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
    at android.widget.AdapterView.performItemClick(AdapterView.java:315)
    at android.widget.AbsListView.performItemClick(AbsListView.java:1205)
    at android.widget.AbsListView$PerformClick.run(AbsListView.java:3310)
    at android.widget.AbsListView$3.run(AbsListView.java:4371)
    at android.os.Handler.handleCallback(Handler.java:836)
    at android.os.Handler.dispatchMessage(Handler.java:103)
    at android.os.Looper.loop(Looper.java:203)
    at android.app.ActivityThread.main(ActivityThread.java:6297)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1084)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:945)
```

**Comment by almic** [comment]
Welp, that's a confirmed bug. Just doing "Add Entry" and then "Remove Entry" crashed the example app!

**Comment by RobertZagorski** [comment]
Turns out, that PR https://github.com/PhilJay/MPAndroidChart/pull/4019 fixes the issue.

**Comment by almic** [comment]
Now totally fixed with d67ea481af1e8528c617dfd404d0c7827b0134b2
